src/Chapter4.hs

```diff
@@ -489,6 +502,13 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+  pure :: a -> List a
+  pure x = Cons x Empty
+
+  (<*>) :: List (a -> b) -> List a -> List b
+  Empty <*> _ = Empty
+  _ <*> Empty = Empty
```

Comment:
The case when none of the lists is empty is missing, so it will lead to the runtime error.
The compiler warns you about non-exhaustive pattern matching 🙂